### PR TITLE
[Cardano]: support NFT on Cardano

### PR DESCRIPTION
**Pull request by satoshiotomakan**
## Description

Cardano NFTs are supported in wallet-core. This pull request adds multiple samples (tests) in C++, Kotlin and Swift to explain how to generate and sign an NFT transfer transaction.

## Improvements

This pull request also adds the `TWCardanoOutputMinAdaAmount` FFI that should be used instead of `TWCardanoMinAdaAmount` as it returns a more accurate value for the minimum output amount.

## How to test

```shell
./tools/build-and-test
./tools/ios-test
./tools/android-test
```

## Checklist

- [x] Add multiple samples (tests) in C++, Kotlin and Swift.
- [x] Add `TWCardanoOutputMinAdaAmount`, an alternative to `TWCardanoMinAdaAmount`, with the same behaviour as [cardano-serialization-lib::min_ada_for_output](https://github.com/Emurgo/cardano-serialization-lib/blob/78184e0a2c207c2f8bba57b0d3c437f4c808c125/rust/src/utils.rs#L1424)
- [x] Cover `TWCardanoOutputMinAdaAmount` with Kotlin, Swift tests.

### CODE REVIEW

File: src/Cardano/Transaction.h [hunk 1]
```diff
@@ -103,6 +103,9 @@ class TxOutput {
     /// Token amounts (optional)
     TokenBundle tokenBundle;
 
+    /// Returns minimal amount of ADA for the output.
+    uint64_t minAdaAmount(uint64_t coinsPerUtxoByte) const;
```
Comment (Milerius) [inline src/Cardano/Transaction.h:107]: this function need noexcept specifier.

File: swift/Tests/Blockchains/CardanoTests.swift [hunk 2]
```diff
@@ -221,4 +221,66 @@ class CardanoTests: XCTestCase {
         let txid = output.txID
         XCTAssertEqual(txid.hexString, "6dcf3956232953fc25b8355fb1ded1e912b5802090fd21434d789087d6329683")
     }
+    
+    func testSignNftTransfer() throws {
```
Comment (Milerius) [inline swift/Tests/Blockchains/CardanoTests.swift:225]: Let's also add the URL of the transaction here

File: tests/chains/Cardano/SigningTests.cpp [hunk 3]
```diff
@@ -15,6 +15,7 @@
 #include "uint256.h"
 #include "TestUtilities.h"
 #include <TrustWalletCore/TWAnySigner.h>
+#include "HDWallet.h"
```
Comment (Milerius) [inline tests/chains/Cardano/SigningTests.cpp:18]: not required

**Review by Milerius — APPROVED** [review]
Good, minor comments.

**Review by Milerius — APPROVED** [review]

**Review by lamafab — APPROVED** [review]

**Pull request merged**